django/db/models/options.py

```diff
@@ -354,6 +336,21 @@ def swapped(self):
                     return swapped_for
         return None
 
+    @property
```

Comment:
The problem with `Manager.contribute_to_class` is that we wouldn't be able to reach the cache of the inheriting children.